booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -524,11 +524,19 @@ private static void generateScenarioRelatedChecks(List<JsonContentCheck> checks,
       "carrierServiceName",
       delayedValue(cspSupplier, CarrierScenarioParameters::carrierServiceName)
     ));
-    checks.add(JsonAttribute.mustEqual(
+
+    checks.add(JsonAttribute.customValidator(
       "[Scenario] Verify that the correct 'contractQuotationReference' is used",
-      "contractQuotationReference",
-      delayedValue(cspSupplier, CarrierScenarioParameters::contractQuotationReference)
-    ));
+      body -> {
```

Comment:
Should we have in the Scenario message also correct 'contractQuotationReference'/'serviceContractReference' is used?